include/bluetooth/mesh/model_types.h

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+/**
+ * @file
+ * @defgroup bt_mesh_model_types Common model types
+ * @{
+ * @brief Collection of types exposed in the model definitions.
+ */
+
+#ifndef BT_MESH_MODEL_TYPES_H__
+#define BT_MESH_MODEL_TYPES_H__
+
+#include <bluetooth/mesh.h>
+
+/** Maximum permissible transition time in milliseconds */
+#define BT_MESH_MODEL_TRANSITION_TIME_MAX_MS (K_MINUTES(10) * (0x3e))
+
+/** Generic Transition parameters for the model messages. */
+struct model_transition {
+	u32_t time; /**< Transition time value in milliseconds */
+	u32_t delay; /**< Message execution delay in milliseconds */
+};
+
+/**
+ * Transaction ID structure, storing information about the previous
+ * transaction in model spec messages.
+ */
+struct transaction_id {
+	u16_t src;
+	u16_t dst;
+	u64_t timestamp;
+	u8_t tid;
+};
+
+/**
+ * Acknowledged message context for tracking the status of model messages
+ * pending a response.
+ */
+struct model_ack_ctx {
+	struct k_sem sem; /**< Sync semaphore. */
+	u32_t op; /**< Opcode we're waiting for. */
+	u16_t dst; /**< Address of the node that should respond. */
+	void *param; /**< User specific parameter. */
+};
+
+enum model_status {
+	/** Command successfully processed. */
+	MODEL_SUCCESS,
+	/** The provided value for range min cannot be set. */
+	MODEL_ERROR_INVALID_RANGE_MIN,
+	/** The provided value for range max cannot be set. */
+	MODEL_ERROR_INVALID_RANGE_MAX,
+	/** Invalid status code. */
+	MODEL_STATUS_INVALID,
+};
+
+/** @cond INTERNAL_HIDDEN
+ * @def MODEL_USER_DATA
+ *
+ * @brief Internal utility macro for type checking model user data.
```

Comment:
Does it make sense to add `@brief` for hidden?
The explanation looks fine. just wondering about the doxygen syntax.

Reply:
I suppose not, other than for following convention.